tests/ui/unit/filtersSpec.js

```diff
@@ -1,19 +1,245 @@
+          .toEqual('How about (<strong>some</strong>) parentheses');
+    });
+
+    it('highlights common terms with slashes', function() {
+        var highlightCommonTerms = $filter('highlightCommonTerms');
+        expect(highlightCommonTerms('This /string/has/a lot\\of\\slashes', 'Short string of words'))
+          .toEqual('This /<strong>string</strong>/has/a lot\\<strong>of</strong>\\slashes');
+    });
+
+    it('highlights common terms with colons', function() {
+        var highlightCommonTerms = $filter('highlightCommonTerms');
+        expect(highlightCommonTerms('What if I assign property:value somewhere?', 'I value your opinion'))
+          .toEqual('What if <strong>I</strong> assign property:<strong>value</strong> somewhere?');
+    });
+});
+
+describe('escapeHTML filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('escapes some html symbols', function() {
+        var escapeHTML = $filter('escapeHTML');
+        expect(escapeHTML('<This \'is the \"worst>')).toEqual('&lt;This &#39;is the &quot;worst&gt;');
+    });
+});
+
+describe('getRevisionUrl filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('escapes some html symbols', function() {
+        var getRevisionUrl = $filter('getRevisionUrl');
+        expect(getRevisionUrl('1234567890ab', 'mozilla-inbound'))
+          .toEqual('/#/jobs?repo=mozilla-inbound&revision=1234567890ab');
+    });
+});
+
+describe('showOrHide filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('says show', function() {
+        var showOrHide = $filter('showOrHide');
+        expect(showOrHide(' me', true)).toEqual('show me');
+    });
+
+    it('says hide', function() {
+        var showOrHide = $filter('showOrHide');
+        expect(showOrHide(' me', false)).toEqual('hide me');
+    });
+});
+
+describe('stripHtml filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('deletes html tags', function() {
+        var stripHtml = $filter('stripHtml');
+        expect(stripHtml('My <html is> deleted')).toEqual('My  deleted');
+    });
+});
+
+describe('linkifyClassifications filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('linkifies classifications', function() {
+        var linkifyClassifications = $filter('linkifyClassifications');
+        // This keeps saying "repo is null"...
+        //expect(linkifyClassifications('This 1234567890ab is linked', 'try')).toEqual('');
```

Comment:
I think $rootScope.repos needs to be defined for this to work:

https://github.com/mozilla/treeherder/blob/master/ui/js/models/repository.js#L18